### ISSUE TITLE
Top Level Navigation Options

### ISSUE THREAD

**Issue by joepavitt** [flowforge/website#319]
### Description

With @iskerrett coming onboard, I wanted to do a review of our top-level navigation options. We have a lot of content that is only accessible via Footer links, and whilst some of this may not be too valuable, it does add flavour and personality to our site, e.g. handbook/about us/team.

#### Header

<img width="1030" alt="Screenshot 2023-01-10 at 14 02 55" src="https://user-images.githubusercontent.com/99246719/211571553-32288ea9-40a3-47fb-abe3-572fb1ab7c76.png">

#### Footer

<img width="1018" alt="Screenshot 2023-01-10 at 14 03 07" src="https://user-images.githubusercontent.com/99246719/211571599-86a96282-2a01-419b-9051-c4a83fdf154d.png">

### Acceptance Criteria

- [ ] Agree upon most important navigation links
- [ ] Produce designs to show alternatives
- [ ] Develop/Deploy alternatives

**Comment by joepavitt** [comment]
Advantage of what we have now is that they're all single-click links, no dropdowns. Traffic shows heavy flow towards pricing/features/docs:

<img width="1027" alt="Screenshot 2023-01-10 at 14 05 47" src="https://user-images.githubusercontent.com/99246719/211572228-6775297a-304f-46fc-b63e-cd04e86a94c3.png">

If we introduce more links (i.e. company links) - they would overpopulate that space, and require dropdowns. Perfectly common to see, but for the purpose of introducing less important links, do we want to make the existing links two-clicks?

**Comment by joepavitt** [comment]
As a quick thought, something like this could work:

<img width="1048" alt="Screenshot 2023-01-10 at 14 31 05" src="https://user-images.githubusercontent.com/99246719/211578474-1a686725-e6b0-4633-9f9c-593c3e3a7421.png">

<img width="1342" alt="Screenshot 2023-01-10 at 14 31 13" src="https://user-images.githubusercontent.com/99246719/211578501-746f7504-e841-4b52-b102-8e4ddaed7b12.png">

**Comment by joepavitt** [comment]
could even add CTAs in for hte "Product" dropdown - although it then starts getting a little busier, so may want to re-think the above design

**Comment by iskerrett** [comment]
I agree we need to look at this but I would like to wait a couple of weeks. I want to get a better handle on how we will ramp up marketing. I hope this is okay.

**Comment by joepavitt** [comment]
> I agree we need to look at this but I would like to wait a couple of weeks

Absolutely, just putting this on the radar and with some initial thinking to make sure it gets addressed.

**Comment by iskerrett** [comment]
I spent some time thinking about this and looking at other websites. Here is the hierarchy I think would work well for us.
Home

- Platform   (over time this might need to become a drop down)
- Solutions  (not initially but soon)
- Resources
-- Node-RED
--- Getting Started
--- GitHub
--- Docs
--- Support Forums
-- FlowForge
--- Getting Started
--- GitHub
--- Docs
--- Support Forums  (where do FlowForge customers ask support questions?)
-- Blog
-- Webinars
-- Newsletter

- Pricing
- Company
-- About
-- Blog
-- Team
-- Jobs
-- Handbook
-- Request Support
-- Contact US
-- Privacy
- Contact Us
- Get Free Trial   (ideally this is a button in the nav bar)
Footer
>> Repeat top navigation hierarchy in the footer
- Plus add Social

**Comment by joepavitt** [comment]
I prefer "Features" over "Platform", as we only have the one platform at the moment, and "features" is a more inviting/informative label to click.

Little hesitant at the overload of Resources, although I think that's also because it's such a big difference from where we are, and we don't necessarily have enough "webinar" content to fill a whole page right now, nor will we for a little while.

Love the idea of having a Node-RED dedicated section, and us producing more dedicated Node-RED resources like "Getting Started Guides" and "Tutorials".

As a starting point, I'd like to extend our blog to be category/filtering based, so we populate all of our content in one place, but make it possible to have the same screen sharable with filters applied (that can also be controlled on-screen). I was hoping to get to making some designs for this yesterday, so that I can articulate my thinking better, but got bogged down with PostHog replication failures and the Team Library UI dev work.

RE: Free Trial as top level action, happy for us to have this, _if_ our new primary CTA is getting members onto FF Cloud, and not GitHub installs. I'm fine either way, just want to make sure the latter is not overlooked.

**Comment by iskerrett** [comment]
> I prefer "Features" over "Platform", as we only have the one platform at the moment, and "features" is a more inviting/informative label to click.

Features is fine for me too.

> 
> Little hesitant at the overload of Resources, although I think that's also because it's such a big difference from where we are, and we don't necessarily have enough "webinar" content to fill a whole page right now, nor will we for a little while.
> 

The main CTA on the webinar page is actually promoting registration for the upcoming webinar. Below will be space for the past webinars.


> Love the idea of having a Node-RED dedicated section, and us producing more dedicated Node-RED resources like "Getting Started Guides" and "Tutorials".
> 
> As a starting point, I'd like to extend our blog to be category/filtering based, so we populate all of our content in one place, but make it possible to have the same screen sharable with filters applied (that can also be controlled on-screen). I was hoping to get to making some designs for this yesterday, so that I can articulate my thinking better, but got bogged down with PostHog replication failures and the Team Library UI dev work.

OK

> 
> RE: Free Trial as top level action, happy for us to have this, _if_ our new primary CTA is getting members onto FF Cloud, and not GitHub installs. I'm fine either way, just want to make sure the latter is not overlooked.

I believe our KPI is number of managed instances. FF Cloud is going to be the easiest way to get people to use FF.  We should only have one CTA at the top, hence my desire to have the Free Trial.

**Comment by joepavitt** [comment]
With the request in #326 we would be updating the existing "Blog" navigation option to be "Resources", then expand that (currently) to "Blog" and "Webinars"

**Comment by joepavitt** [comment]
Just to record interaction with top-level navigation as things stand:

<img width="1054" alt="Screenshot 2023-03-01 at 16 46 33" src="https://user-images.githubusercontent.com/99246719/222206106-9e828326-9aef-413c-96f9-7d6b1a3f9435.png">

**Comment by joepavitt** [comment]
Curious to understand what we need to focus on here, the top-level options we have now align with those requested in https://github.com/flowforge/website/issues/319#issuecomment-1382027701 hence the issue being closed.

The only differences exist within "Resources" where we could have a lot more content - is that where you want the focus @iskerrett?

**Comment by iskerrett** [comment]
We need to expand the resource menu to the following:
Resources
-- Node-RED
--- Getting Started
--- GitHub
--- Docs
--- Support Forums
-- FlowForge
--- Getting Started
--- GitHub
--- Docs
--- Support Forums (where do FlowForge customers ask support questions?)
-- Blog
-- Webinars
-- Newsletter


For the company page, the order of the menu items needs to be updated:
Company
-- About
-- Blog
-- Team
-- Jobs
-- Handbook
-- Request Support
-- Contact US
-- Privacy

**Comment by iskerrett** [comment]
> for each of the following, just want to be sure we're on same page on where we are sending the traffic:
> 
> * Node-RED
>   
>   * Getting Started - https://nodered.org/docs/getting-started/

https://flowforge.com/blog/2023/01/getting-started-with-node-red/

>   * GitHub - https://github.com/node-red/node-red
>   * Docs - https://nodered.org/docs/
>   * Support Forums - https://discourse.nodered.org/
> * FlowForge
>   
>   * Getting Started - Install Docs? Getting started with FF Cloud?

it seems we need to create a Getting Started doc. 

>   * GitHub - https://github.com/flowforge
>   * Docs - https://flowforge.com/docs/
>   * Support Forums - Where do we want to point people?

Let's delete this.

**Comment by iskerrett** [comment]
Thinking a bit more about the Company menu item. I would remove the Handbook and have it accessible from the footer.

**Comment by joepavitt** [comment]
Just to check @iskerrett - you want "blog" appearing under _both_ "company" and "resources?" in the top-level nav?

**Comment by joepavitt** [comment]
Quick first draft for "Resources":

<img width="1902" alt="Screenshot 2023-03-24 at 16 59 19" src="https://user-images.githubusercontent.com/99246719/227591674-677a0ae6-95f8-4564-8dbf-cd3a3afa2100.png">

**Comment by iskerrett** [comment]
> @iskerrett just to [chase on my above question](https://github.com/flowforge/website/issues/319#issuecomment-1483040225) - are we going for "blog" in _both_ Company and Resources list?

Yes please. 
> 
> Then to chase on the "Getting Started" for FlowForge, am I just leaving that out for now, and just having the "GitHub" and "Docs" links for now?

Yes

**Comment by Yndira-FlowForge** [comment]
Hi, [here's a prototype](https://www.figma.com/proto/hXDN5PnY3TmeQSv3n7Hna5/FlowForge-Website?page-id=0%3A1&node-id=1278-6778&viewport=-182%2C1405%2C0.14&scaling=min-zoom&starting-point-node-id=804%3A2591).

All of the menu items (that have a sub-menu) should expand when you hover, except for "resources", where you'll have to click to see it. I did it like this so you can also hover on the items of that sub-menu and see the color change, but on the website, it should behave like the rest of them.

If you scroll, you'll see the nav-bar is fixed at the top, as we discussed in our call.

Since the idea is to show the sub-menus on hover, I removed the arrow next to the item that suggested to click to reveal more options, I think those are expandable and this way we can save some space. 

I added icons to make it easier to navigate and a bit friendlier.

I'll have to work on the mobile version, and maybe improve the transitions, but wanted some feedback on the layout first.

Let me know what you think :)

**Comment by iskerrett** [comment]
> All of the menu items (that have a sub-menu) should expand when you hover, except for "resources", where you'll have to click to see it. I did it like this so you can also hover on the items of that sub-menu and see the color change, but on the website, it should behave like the rest of them.

I like it. Looks very clean. The Resource sub menu I have some feedback but I am assuming you just want feedback now on the main nav.

> 
> If you scroll, you'll see the nav-bar is fixed at the top, as we discussed in our call.

Great idea. Very nice
> 
> Since the idea is to show the sub-menus on hover, I removed the arrow next to the item that suggested to click to reveal more options, I think those are expandable and this way we can save some space.

Good idea.

> 
> I added icons to make it easier to navigate and a bit friendlier.

I really like these :-)

**Comment by Yndira-FlowForge** [comment]
Hi @iskerrett I'm glad you like it! Share your feedback on the resources sub-menu as well, please.
@joepavitt what do you think?

**Comment by joepavitt** [comment]
> If you scroll, you'll see the nav-bar is fixed at the top, as we discussed in our call.

I like the idea of sticky header, although we should make sure it's consistent padding above/below, looks odd with barely any padding below, but a lot above.

Could also get away with removing a bit of vertical padding on the "Node-RED/FlowForge" bar and the bottom bar of three options, should be a consistent row-height to the main contents here.

> I removed the arrow next to the item that suggested to click to reveal more options

Whilst this does remove space (great) it does now offer an inconstancy in our header, some you click, some over hover, and no indication in advance of which are which. Having had a look around though, seems like this is fairly standard (see [Stripe](https://stripe.com/en-gb-nl)) - so I'll step back on my thoughts here and say - that's fine :)

> I added icons to make it easier to navigate and a bit friendlier.

Looks better :+1: may want a different icon for "blog" though, as pencil generally means "edit".

Just also want to check as I know Ian wasn't a fan of the three items being at the bottom, seems like you've discussed this already, just want to be sure we're definitely. not going with a 3-column approach?

**Comment by iskerrett** [comment]
@Yndira-FlowForge the challenge I have with the current design of the resource menu is that it treats different items differently. Why are the three items at the bottom and not in a list? What happens when we need to add another menu item?

For how the columns are treated, I wonder if it would be better to have the dividing line go up between Node-RED and FlowForge? Right now I find the column headings and the lists don't really match up very well.

@joepavitt not sure why you object to a 3 column approach to the floating menu? Can you explain?

**Comment by joepavitt** [comment]
I didn't object, I was bringing it up for that reason. I thought I remember you saying that you wanted that, but it wasn't reflected in the latest design which you seemed to be happy with.

Missed that you hadn't explicitly fedback on the floating menu.

**Comment by joepavitt** [comment]
I didn't object, I was bringing it up for that reason. I thought I remember you saying that you wanted that, but it wasn't reflected in the latest design which you seemed to be happy with.

Missed that you hadn't explicitly fedback on the floating menu.

**Comment by iskerrett** [comment]
Ok, good. It actually might make more sense to have just two columns: Node-RED and FlowForge.  Under FlowForge have the following:
Getting Started
github
docs 
blog
webinar 
newsletter


Otherwise, the three column would be:
FlowForge
- getting started
- github
- docs

Other
- blog
- webinar
- newsletter

**Comment by Yndira-FlowForge** [comment]
Hi, my thinking was the same as Joe, I thought "others" wasn't meaningful enough to deserve a "title" at the same level as "FlowForge" and "Node-RED"

**Comment by iskerrett** [comment]
I am open to suggestion for something than Other?

**Comment by joepavitt** [comment]
> I am open to suggestion for something than Other?

Yeah, I struggled with thinking around this. I went down the "Events"/"Content"/"Articles" type route, but none of them were a suitable catch-all.

Could try a two-column approach within the "FlowForge" section? So it's 3 columns, but designed such that the "other" bits fall within the FlowForge part?

**Comment by iskerrett** [comment]
Good idea. The 3 other items are really part of FlowForge so they fit there.

**Comment by Yndira-FlowForge** [comment]
> I like the idea of sticky header, although we should make sure it's consistent padding above/below, looks odd with barely any padding below, but a lot above.

I centered the content of the header. I didn't do this initially because now the items are a bit closer to the top than in the original design and didn't want to change that. Also, the distance between the sub-menu and the menu item needs to be bigger and I wanted to keep them a bit closer. It's not as bad as I thought it would be, though.

Here it is with all the revisions (updated the prototype as well)
<img width="1440" alt="Screenshot 2023-04-18 at 16 04 10" src="https://user-images.githubusercontent.com/129537638/232807859-db4a037d-4670-4623-b1a9-b3c05bc08776.png">

And, here's one with only two columns, in case you think the one above is too wide
<img width="1440" alt="Screenshot 2023-04-18 at 16 03 02" src="https://user-images.githubusercontent.com/129537638/232808091-77b87737-46ee-424f-8a8e-ca02cf46ff5d.png">

**Comment by Yndira-FlowForge** [comment]
@joepavitt I tried it like that because of Ian's comment

> For how the columns are treated, I wonder if it would be better to have the dividing line go up between Node-RED and FlowForge? Right now I find the column headings and the lists don't really match up very well.

I didn't want to make the line go all the way up, it doesn't look good...

This could be a workaround
![Image](https://user-images.githubusercontent.com/129537638/232828933-6a2e60e9-f536-4f2c-a029-ac6adbcec589.png)

**Comment by iskerrett** [comment]
I agree the 3 column looks best.

fwiw, the support forums item should be under Node-RED.

Nice work Yrinda

**Comment by Yndira-FlowForge** [comment]
You're right @iskerrett
Here it is
<img width="1441" alt="Screenshot 2023-04-18 at 18 00 28" src="https://user-images.githubusercontent.com/129537638/232835581-01e86964-272d-4973-80c9-fb84baeb91da.png">

**Comment by Yndira-FlowForge** [comment]
I've been working on the mobile version.

Currently it's like this:
<img width="393" alt="Screenshot 2023-04-21 at 12 28 27" src="https://user-images.githubusercontent.com/129537638/233614146-74eb5822-b496-4eee-8222-2ad19861cd8b.png">

I noticed that the sign in / try for free options are not visible in the mobile version and it has a light background. 
In this proposal I've maintained the same dark background of the header and added a small animation to it, and included two buttons for the missing options. It would look like this:
 
<img width="1929" alt="Screenshot 2023-04-21 at 12 27 38" src="https://user-images.githubusercontent.com/129537638/233614764-f5e728d1-dfa1-46bf-a15f-2cd565f2c912.png">

[Here's a prototype](https://www.figma.com/proto/hXDN5PnY3TmeQSv3n7Hna5/FlowForge-Website?page-id=0%3A1&node-id=1406-10994&viewport=-6893%2C8373%2C1&scaling=min-zoom&starting-point-node-id=1406%3A10994&show-proto-sidebar=1), the only interactive options are the ones showed in the mockups (product, resources>FlowForge). I added a scroll bar for the menu that overflows the screen for reference, but it doesn't have the animation.
